### PR TITLE
Add support for PostgreSQL

### PR DESCRIPTION
**Pull request by subaru9**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ gem 'rails-i18n'
 gem 'responders', '~> 2.0'
 
 gem 'puma'
-gem 'mysql2', '~> 0.3.21'
+gem 'pg', '~> 0.21'
```
Comment (yivo) [inline Gemfile]: @mitjok The task is not to replace MySQL but support PG as alternate DB.

**Review by yivo — CHANGES_REQUESTED** [review]

**Comment by subaru9** [comment]
@yivo fixed

File: config/initializers/abstract_mysql2_adapter.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+return unless File.exists?('active_record/connection_adapters/mysql2_adapter.rb')
```
Comment (yivo) [inline config/initializers/abstract_mysql2_adapter.rb]: @mitjok It is better to check for ruby constant.

File: app/models/account_version.rb [hunk 3]
```diff
@@ -52,8 +52,14 @@ def self.optimistically_lock_account_and_create!(balance, locked, attrs)
     select  = Account.unscoped.select(values).where(id: account_id, balance: balance, locked: locked).to_sql
     stmt    = "INSERT INTO account_versions (#{columns}) #{select}"
 
-    connection.insert(stmt).tap do |id|
-      if id == 0 || id.blank? # if record not inserted mysql returns id = 0 and mssql returns id = nil
+    # Dmytro LitsLink notes on PostgreSql 10.2: 
+    # 1. ActiveRecord::ConnectionAdapters::DatabaseStatements#insert 
+    # returns nil if account version can't be created. 
+    # 2. Need to specify primary key column name explicitly 
+    # to produce sql with `RETURNING "id"`at the end of the `INSERT` SQL statement
```
Comment (yivo) [inline app/models/account_version.rb]: @mitjok space is missing before `at`.

File: app/models/account_version.rb [hunk 4]
```diff
@@ -52,8 +52,14 @@ def self.optimistically_lock_account_and_create!(balance, locked, attrs)
...
+    # 3. mssql returns id = nil
```
Comment (yivo) [inline app/models/account_version.rb]: @mitjok Can you please use `MSSQL`?

**Review by yivo — CHANGES_REQUESTED** [review]

File: app/models/account_version.rb [hunk 5]
```diff
@@ -26,7 +26,7 @@ class AccountVersion < ActiveRecord::Base
 
   scope :history, -> { with_reason(*HISTORY).reverse_order }
 
-  # Use account balance and locked columes as optimistic lock column. If the
+  # Use account balance and locked columnes as optimistic lock column. If the
```
Comment (yivo) [inline app/models/account_version.rb]: @mitjok It should be `columns`.

**Review by yivo — CHANGES_REQUESTED** [review]
@mitjok Can you please include guide on how to ge PG with Peatio at `docs/databases/postgresql.md`.

**Comment by subaru9** [comment]
@yivo not sure why this build is [failing](https://www.evernote.com/l/AYMiBDb7gTxI_I53vEKa_tPeQbfyfIAn6XAB/image.png)

**Review by yivo — APPROVED** [review]

**Pull request merged**